src/main/java/com/iotiq/application/repository/ProductDemandRepository.java

```diff
@@ -17,7 +17,7 @@ public interface ProductDemandRepository extends JpaRepository<ProductDemand, UU
 
     Page<ProductDemand> findAllByCustomerAndIsActiveTrue(Customer customer, Pageable pageable);
     Optional<ProductDemand> findByIdAndCustomerAndIsActiveTrue(UUID id, Customer customer);
-    Page<ProductDemand> findAllByIsActiveTrueAndSellerIsNullAndSustainabilitySkillsIn(Pageable pageable, Set<Skill> skills);
+    Page<ProductDemand> findByIsActiveTrueAndSellerIsNullAndSustainabilitySkillsEmptyOrSustainabilitySkillsIn(Pageable pageable, Set<Skill> skills);
```

Comment:
This is wrong in my opinion. Here is the query it generates:

`select p from ProductDemand p left join p.sustainability.skills skills where p.isActive = true and p.seller is null and p.sustainability.skills is empty or skills in :pageable`

I think qe should write the query with Query annotation, pereferably a jpql query.

Reply:
This is correct. But you need to add a test including a seller with a set of skills and verify that product demands with those skills are included in the result. (You can do it in another pr)